app/code/Magento/CustomerGraphQl/Model/Resolver/RequestPasswordResetEmail.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CustomerGraphQl\Model\Resolver;
+
+use Magento\Customer\Api\AccountManagementInterface;
+use Magento\Customer\Model\AccountManagement;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Framework\Validator\EmailAddress as EmailValidator;
+
+/**
+ * Class Resolver for RequestPasswordResetEmail
+ */
+class RequestPasswordResetEmail implements ResolverInterface
+{
+    /**
+     * @var AccountManagementInterface
+     */
+    protected $customerAccountManagement;
+
+    /**
+     * @var EmailValidator
+     */
+    protected $emailValidator;
+
+    /**
+     * RequestPasswordResetEmail constructor.
+     *
+     * @param AccountManagementInterface $customerAccountManagement
+     * @param EmailValidator             $emailValidator
+     */
+    public function __construct(
+        AccountManagementInterface $customerAccountManagement,
+        EmailValidator $emailValidator
+    ) {
+        $this->customerAccountManagement = $customerAccountManagement;
+        $this->emailValidator = $emailValidator;
+    }
+
+    /**
+     * @param Field            $field
+     * @param ContextInterface $context
+     * @param ResolveInfo      $info
+     * @param array|null       $value
+     * @param array|null       $args
+     *
+     * @return bool
+     * @throws GraphQlInputException
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (empty($args['email'])) {
+            throw new GraphQlInputException(__('Email must be specified'));
+        }
+
+        if (!$this->emailValidator->isValid($args['email'])) {
+            throw new GraphQlInputException(__('Email is invalid'));
+        }
+
+        try {
+            return $this->customerAccountManagement->initiatePasswordReset(
+                $args['email'],
+                AccountManagement::EMAIL_RESET
+            );
+        } catch (\Exception $e) {
```

Comment:
Just curious. Is there some unhandled exception in the flow? If it is so, we should not simply return `false` but throw `LocalizedException` instead. 

But the main question is, do we have an unhandled exception somewhere in the middle?

Reply:
Added `LocalizedException` .
Yes,  we have exceptions inside `$this->customerAccountManagement->resetPassword()` and `$this->customerAccountManagement->initiatePasswordReset()`  these exceptions related with 

1. Check if customer account is for this email
2. Check match token 
3. Check length of password